### PR TITLE
Add log rotation

### LINKED ISSUE #139: log/sitebuilder.log raised > 2Go, should use rotation. Too big to be used

### PR DESCRIPTION
**Pull request by t4deu**
Because without the log rotation the file becomes very large to debug, a log file per
day is better to handle.

Also Fix the fatal error that happens sometimes on deploy when try to write a log.

closes #139 and #54

### CODE REVIEW

**Comment by juliogreff** [comment]
This creates smaller files, but it doesn't fix the problem that we have just too many of them. I'd rather have a single file that's bounded to a certain limit.

**Comment by t4deu** [comment]
Monolog has an [rotation handler](https://github.com/Seldaek/monolog/blob/master/src/Monolog/Handler/RotatingFileHandler.php) that accepts amount of files limit, we could use that?

**Comment by juliogreff** [comment]
```
 * This rotation is only intended to be used as a workaround. Using logrotate to
 * handle the rotation is strongly encouraged when you can use it.
```

**Comment by t4deu** [comment]
Ok, so with logrotade config will be like, one per server:
```
# /etc/logrotate.d/sitebuilder
/meumobi/PROJECTS/meumobi.com/shared/log/*.log {
   rotate 12 # max of 12 files
   weekly # log file per week
   missingok
   notifempty
   compress
   delaycompress
   create 777 meumobi meumobi # file permission and onwer
}
```

**Comment by juliogreff** [comment]
Let's keep a single file for each current log, so you will need to adjust your pull request. Also, try to keep config in the repo, and symlink.

**Comment by t4deu** [comment]
The changes made and this branch will be removed, since they are not needed anymore.

The logrotate config that will be used is the previous described:
```
/meumobi/PROJECTS/meumobi.com/shared/log/*.log {
   rotate 12 # max of 12 files
   weekly # log file per week
   missingok
   notifempty
   compress
   delaycompress
   create 777 meumobi meumobi # file permission and onwer
}
```

The sitebuilder setup wiki need to be updated to explain how to setup the logrotate.
And this configuration can be printed by the capistrano "setup" task to explain how to configure the logrotate.

**Comment by juliogreff** [comment]
Why not create the file inside the repo and symlink? It's less room for error, and easier for us to update.

**Comment by t4deu** [comment]
Yes the file will be in the repo, sorry for not say that, but the wiki need to be updated.
The file path will be:
> config/logrotate.sample

And I thought it would be good that after the instance setup, print the hint explaining how to configure the logrotate, but this is optional

**Comment by t4deu** [comment]
Can't use a symlink for the logrotate config, the file owner need to be "root" to logrotate work properly otherwise this error happens:
> Ignoring config/logrotate because the file owner is wrong (should be root).

For now we can manually setup the logrotate on each server.

**Comment by juliogreff** [comment]
Either way, keep it on the repo, so we just copy the file instead of relying on the wiki

**Comment by t4deu** [comment]
ok, already added the file

File: config/logrotate.sample [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/home/meumobi/PROJECTS/int-meumobi.com/shared/log/*.log {
```
Comment (juliogreff) [inline config/logrotate.sample]: This only works for INT. What about the other environments?

Comment (t4deu) [inline config/logrotate.sample]: It's a sample

Comment (juliogreff) [inline config/logrotate.sample]: Again, it's margin for error. I'd rather have a sample for every environment, and then it's just a copy and paste, with no editing at all.

**Comment by juliogreff** [comment]
Why do you have a .sample file, and the other have no such extension?

**Comment by juliogreff** [comment]
I'm merging the PR. Don't forget to close the issue once the config is done on the servers. Indicate on the ticket what you've done.

**Pull request merged**